pkg/virt-handler/vm_test.go

```diff
@@ -1701,6 +1700,181 @@ var _ = Describe("VirtualMachineInstance", func() {
 	})
 })
 
+var _ = Describe("DomainNotifyServerRestarts", func() {
+	Context("should establish a notify server pipe", func() {
+		var shareDir string
+		var err error
+		var serverStopChan chan struct{}
+		var serverIsStoppedChan chan struct{}
+		var stoppedServer bool
+		var domainPipeStopChan chan struct{}
+		var stoppedPipe bool
+		var eventChan chan watch.Event
+		var client *notifyclient.Notifier
+		var recorder *record.FakeRecorder
+		var vmiStore cache.Store
+
+		BeforeEach(func() {
+			serverStopChan = make(chan struct{})
+			domainPipeStopChan = make(chan struct{})
+			serverIsStoppedChan = make(chan struct{})
+			eventChan = make(chan watch.Event, 100)
+			stoppedServer = false
+			stoppedPipe = false
+			shareDir, err = ioutil.TempDir("", "kubevirt-share")
+			Expect(err).ToNot(HaveOccurred())
+
+			recorder = record.NewFakeRecorder(10)
+			vmiInformer, _ := testutils.NewFakeInformerFor(&v1.VirtualMachineInstance{})
+			vmiStore = vmiInformer.GetStore()
+
+			go func(serverIsStoppedChan chan struct{}) {
+				notifyserver.RunServer(shareDir, serverStopChan, eventChan, recorder, vmiStore)
+				close(serverIsStoppedChan)
+			}(serverIsStoppedChan)
+
+			time.Sleep(3)
+		})
+
+		AfterEach(func() {
+			if stoppedServer == false {
+				close(serverStopChan)
```

Comment:
saw you mentioned about flakiness,
can it be because there isnt wait here for the sever to be actually closed before proceeding ?
if you wait, you might see it takes time (around 120 seconds because the HTTP handshaking timeout we saw once)